src/gui/gtk.c

```diff
@@ -3066,6 +3066,110 @@ GtkWidget *dt_ui_notebook_page(GtkNotebook *notebook, const char *text, const ch
   return page;
 }
 
+static gint _get_container_row_heigth(GtkWidget *w)
+{
+  gint height = DT_PIXEL_APPLY_DPI(10);
+
+  if(GTK_IS_TREE_VIEW(w))
+  {
+    gint cell_height = 0;
+    gtk_tree_view_column_cell_get_size(gtk_tree_view_get_column(GTK_TREE_VIEW(w), 0),
+                                       NULL, NULL, NULL, NULL, &cell_height);
+    GValue separation = { G_TYPE_INT };
+    gtk_widget_style_get_property(w, "vertical-separator", &separation);
+
+    if(cell_height > 0) height = cell_height + g_value_get_int(&separation);
+  }
+  else
+  {
+    GList *children = gtk_container_get_children(GTK_CONTAINER(w));
+    if(children)
+    {
+      height = gtk_widget_get_allocated_height(GTK_WIDGET(children->data));
+      g_list_free(children);
+    }
+  }
+
+  return height;
+}
+
+static void _scroll_wrap_resize(GtkWidget *w, GtkAllocation *allocation, const char *config_str)
+{
+  GtkWidget *sw = gtk_widget_get_parent(w);
+  if(GTK_IS_VIEWPORT(sw)) sw = gtk_widget_get_parent(sw);
+
+  gint increment = _get_container_row_heigth(w);
+
+  gint height = dt_conf_get_int(config_str);
+
+  gint max_height = DT_PIXEL_APPLY_DPI(1000);
+
+  height = (height < 1) ? 1 : (height > max_height) ? max_height : height;
+
+  dt_conf_set_int(config_str, height);
```

Comment:
After my recent fun with dt conf you might replace this whole bit with `dt_conf_get_and_sanitize_int()`

Reply:
We are almost 100% in sync!

Reply:
> After my recent fun with dt conf you might replace this whole bit with `dt_conf_get_and_sanitize_int()`

After the recent conf changes, what do you suggest I do here? Add 
`    <type min="1" max="1000">int</type>
`
to each of the entries in the xml and just do a simple dt_conf_get_int here?

Reply:
Later this weekend I'll propose a PR that will allow just that :)

Reply:
My PR is done now :) also I'd recommend more strict "min". `1` seems a little low for module window height :)

Reply:
It gets rounded up to full row/element heights (depending on font etc). So 1 is intentional; there will always be one full row. I don't see a need to force it to be higher than that, if someone want a 1 row scroll window, why stop them?